lib/OpenQA/Jobs/Constants.pm

```diff
@@ -3,47 +3,64 @@ use Mojo::Base -base;
-    ASSIGNED  => 'assigned'
+    ASSIGNED  => 'assigned',  # job has been sent to worker but worker has not acknowledged yet
+    SETUP     => 'setup',     # worker prepares execution of backend/isotovideo (e.g. waiting for cache service)
+    RUNNING   => 'running',   # backend/isotovideo is running
+    UPLOADING => 'uploading', # remaining test results are uploaded after backend/isotovideo has exited
+    CANCELLED => 'cancelled', # job was cancelled while still being scheduled
+    DONE      => 'done',      # worker reported that the job is no longer running or web UI considers job dead/abandoned
 };
+use constant STATES => (SCHEDULED, ASSIGNED, SETUP, RUNNING, UPLOADING, DONE, CANCELLED);
 
-use constant STATES         => (SCHEDULED, ASSIGNED, SETUP, RUNNING, UPLOADING, DONE, CANCELLED);
-use constant PENDING_STATES => (SCHEDULED, ASSIGNED, SETUP, RUNNING, UPLOADING);
-use constant EXECUTION_STATES => (ASSIGNED, SETUP, RUNNING, UPLOADING);
+# define "meta" states
+use constant PENDING_STATES   => (SCHEDULED, ASSIGNED, SETUP,   RUNNING, UPLOADING);
+use constant EXECUTION_STATES => (ASSIGNED,  SETUP,    RUNNING, UPLOADING);
 use constant PRE_EXECUTION_STATES => (SCHEDULED);        # Assigned belongs to pre execution, but makes no sense for now
 use constant FINAL_STATES         => (DONE, CANCELLED);
+use constant {
+    PRE_EXECUTION => 'pre_execution',
+    EXECUTION     => 'execution',
+    FINAL         => 'final',
+};
 
-# Results
+# define results for the overall job
 use constant {
-    NONE               => 'none',
-    PASSED             => 'passed',
-    SOFTFAILED         => 'softfailed',
-    FAILED             => 'failed',
-    INCOMPLETE         => 'incomplete',                  # worker died or reported some problem
-    SKIPPED            => 'skipped',                     # dependencies failed before starting this job
-    OBSOLETED          => 'obsoleted',                   # new iso was posted
-    PARALLEL_FAILED    => 'parallel_failed',             # parallel job failed, this job can't continue
-    PARALLEL_RESTARTED => 'parallel_restarted',          # parallel job was restarted, this job has to be restarted too
-    USER_CANCELLED     => 'user_cancelled',              # cancelled by user via job_cancel
-    USER_RESTARTED     => 'user_restarted',              # restarted by user via job_restart
-    TIMEOUT_EXCEEDED   => 'timeout_exceeded',            # killed by the worker after MAX_JOB_TIME has been exceeded
+    NONE       => 'none',
+    PASSED     => 'passed',
+    SOFTFAILED => 'softfailed',
+    FAILED     => 'failed',
+    INCOMPLETE => 'incomplete',    # worker died or reported some problem
+    SKIPPED =>
+      'skipped',    # dependencies failed before starting this job (FIXME: clarify weird overlap with CANCELLED state)
+    OBSOLETED => 'obsoleted'
+    , # new iso was posted (FIXME: while the job has already been running, right? otherwise the CANCELLED state would have been used?)
```

Comment:
when new iso is posted, the scheduled jobs will be also obsoleted, right?  If the answer is yes, IMHO,  the running jobs and scheduled jobs those are obsoleted should be set the same state.

Reply:
I've just had a look at the code. Apparently in both cases jobs are set to the state `CANCELLED` and to the result `OBSOLETED`. (And when a job is cancelled by a user it is set to the state `CANCELLED` and to the result `USER_CANCELLED`.) So the `CANCELLED` state is actually used for jobs which have already been started.

Note that the combination `DONE`/`OBSOLETED` is possible, too (though an API call with `newbuild` parameter).

For now, I'll change the FIXME comments accordingly to reflect the current state.